### ISSUE TITLE
out-of-bound-read in jpgicc.c

### ISSUE THREAD

**Issue by treebacker** [mm2/Little-CMS#257]
In source code `utils/jpgicc/jpgicc.c`, function `HandleEXIF`:

Below code:

```c
     			 // offset to Image File Directory (includes the previous 8 bytes)
                ifd_ofs = read32(data, pos, swapBytes);

                // Search the directory for resolution tags
                numEntries = read16(data, ifd_ofs, swapBytes);

```

`ifd_ofs` may be a large number with a crafted file, which results a out bound of read in `read16`.

 In AddressSanitizer:
![3_](https://user-images.githubusercontent.com/44844446/115103511-f92e6980-9f84-11eb-9bd3-ed55e0160c94.png)
In gdb:
![3](https://user-images.githubusercontent.com/44844446/115103514-00ee0e00-9f85-11eb-96d6-eefd51bdf2e5.png)

The crafted file is in the zip:
[bug3.zip](https://github.com/mm2/Little-CMS/files/6328688/bug3.zip)

**Comment by mm2** [comment]
Thanks for reporting.
However, what you are reporting may be a vulnerability of jpgicc, which is a **sample** of how to use the little CMS library and **it is not required or used** in client applications using the library. So, I will try to add some checks as long as that would not affect code readability, which is the primary goal of this sample.

 **Please note any CVE regarding this kind of vulnerabilities would be promptly disputed, as are related with samples and not with the little cms component, which is unaffected**.

**Comment by mm2** [comment]
Fixed per 8564c9b4fbf07d23062bbbbd87d5eb10dd92073f Thanks for reporting
